// @aws-amplify/graphql-api-construct/lib/internal/default-parameters.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultTranslationBehavior = void 0;
/**
 * Defaults which will be used by the construct if overrides are not provided.
 * These should not really be updated post-launch, since they're likely breaking changes for customers.
 * Be sure to document default values when adding new keys in the top level `types` file.
 */
exports.defaultTranslationBehavior = {
    shouldDeepMergeDirectiveConfigDefaults: true,
    disableResolverDeduping: true,
    sandboxModeEnabled: false,
    useSubUsernameForDefaultIdentityClaim: true,
    subscriptionsInheritPrimaryAuth: false,
    populateOwnerFieldForStaticGroupAuth: true,
    suppressApiKeyGeneration: false,
    secondaryKeyAsGSI: true,
    enableAutoIndexQueryNames: true,
    respectPrimaryKeyAttributesOnConnectionField: true,
    enableSearchNodeToNodeEncryption: false,
    enableSearchEncryptionAtRest: false,
    enableTransformerCfnOutputs: false,
    allowDestructiveGraphqlSchemaUpdates: false,
    replaceTableUponGsiUpdate: false,
};
